### PR TITLE
[PiranhaSwift] Update PiranhaSwift to use SwiftSyntax 5.2

### PR DESCRIPTION
**Pull request by mkr-plse**
1. Upgrade PiranhaSwift to use SwiftSyntax 5.2
2. Fixes a couple of refactoring bugs: 
     ` private lazy var xyz = cachedExperiments.isInControlGroup(...)` is correctly refactored. (this was completely discarded earlier)
     `private lazy var xyz = !cachedExperiments.....` is also correctly refactored. (earlier, only the RHS was refactored).

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -20,8 +20,8 @@ matrix:
 
     - language: swift
       os: osx
-      osx_image: xcode11.3
-      env: SWIFT=5.1
+      osx_image: xcode11.6
+      env: SWIFT=5.2
```
Comment (lazaroclapp) [inline .travis.yml:24]: Do we intend to be compatible with multiple Swift versions or only 5.2 exclusively?

File: swift/src/CleanupStaleFlags/StaleFlagCleaner.swift [hunk 2]
```diff
@@ -277,90 +280,91 @@ class XPFlagCleaner: SyntaxRewriter {
         if isDeepCleanPass, let value = deepCleanMap[string(of: node)] {
             return value
         }
-
+
```
Comment (lazaroclapp) [inline swift/src/CleanupStaleFlags/StaleFlagCleaner.swift:283]: Delete empty trailing space?

File: swift/tests/testfile.swift [hunk 3]
```diff
@@ -505,7 +512,7 @@ class SwiftExamples {
     private var shouldDoSomething: Bool {
         return cachedExperiments.isTreated(forExperiment: ExperimentNamesSwift.test_experiment)
     }
-
+
```
Comment (lazaroclapp) [inline swift/tests/testfile.swift:515]: Also trailing space.

File: swift/tests/treated.swift [hunk 4]
```diff
@@ -174,9 +174,10 @@ class SwiftExamples {
         y = true
         y = x
         y = true
-
+        print("2")
+        
         print("treated")
-
+
```
Comment (lazaroclapp) [inline swift/tests/treated.swift:180]: Also trailing space.

**Review by lazaroclapp — APPROVED** [review]
Giving up on understanding this a little bit, to be honest, so I only have some cosmetic nits. A lot seem to be syntax changes between Swift 5.1 and 5.2, correct? Might have been clearer with the SwiftSyntax upgrade being one change and the other two fixes a separate PR. Either way: Stamped 🙂

File: .travis.yml (hunk 1 above)
Comment (mkr-plse) [reply .travis.yml:24]: 5.2 exclusively. PiranhaSwift won't be backward compatible with the 5.1 version anymore.

**Comment by mkr-plse** [comment]
Thanks @lazaroclapp. Will fix the nits in a subsequent PR.

**Pull request merged**